src/panels/lovelace/hui-editor.ts

```diff
@@ -208,19 +192,8 @@ class LovelaceFullConfigEditor extends LitElement {
     this._hashAdded = false;
   }
 
-  private _yamlChanged() {
-    this._hash = this._hashAdded || this.textArea.value.includes("#");
-    if (this._changed) {
-      return;
-    }
-    window.onbeforeunload = () => {
-      return true;
-    };
-    this._changed = true;
-  }
-
-  private get textArea(): HTMLTextAreaElement {
-    return this.shadowRoot!.querySelector("textarea")!;
+  private get codeEditor(): HuiCodeEditor {
+    return this.shadowRoot!.querySelector("hui-code-editor")! as HuiCodeEditor;
```

Comment:
`as X` is not needed if you register the element in the global HTML map.

Reply:
That is what I was missing... I was wondering why I had to do this....